scrapy/spider.py

```diff
@@ -31,11 +31,19 @@ def __init__(self, name=None, **kwargs):
         if not hasattr(self, 'start_urls'):
             self.start_urls = []
 
-    def log(self, message, level=log.DEBUG, **kw):
-        """Log the given messages at the given log level. Always use this
-        method to send log messages from your spider
+    @property
+    def logger(self):
+        logger = logging.getLogger(self.name)
```

Comment:
@dangra: sorry about that, I didn't test the backward compatibility functions too much. I've fixed it and rebased master, can you test it again?

@kmike: hworker sets the logging configuration itself I think, without calling `logging_configure`, that should be the reason why TopLevelFormatter isn't set. I like logging the whole logger name too, but I'd make that change after the module relocation so we don't have to update the log samples in the docs twice.